core/src/transport/choice.rs

```diff
@@ -51,13 +51,21 @@ where
         id: ListenerId,
         addr: Multiaddr,
     ) -> Result<(), TransportError<Self::Error>> {
+        println!("Trying address on {}...", std::any::type_name::<A>());
```

Comment:
Please use the `log` crate and trace level instead.

Reply:
Lets say something like "attempting to dial {addr} using {typename}".

Reply:
Done